localstack/cli/localstack.py

```diff
@@ -37,17 +47,38 @@ def _setup_cli_debug():
     setup_logging()
 
 
+def _publish_cmd_as_analytics_event(cmd: str):
+    event = Event(
+        name="cli_cmd",
+        payload={"cmd": cmd},
+        metadata=EventMetadata(
+            session_id=get_session_id(),
+            client_time=str(datetime.datetime.now()),  # TODO: consider using utcnow()
+        ),
+    )
+    publisher = AnalyticsClientPublisher(AnalyticsClient())
+    publisher.publish([event])
+
+
 @click.group(name="localstack", help="The LocalStack Command Line Interface (CLI)")
 @click.version_option(version=__version__, message="%(version)s")
 @click.option("--debug", is_flag=True, help="Enable CLI debugging mode")
 @click.option("--profile", type=str, help="Set the configuration profile")
 def localstack(debug, profile):
     if profile:
         os.environ["CONFIG_PROFILE"] = profile
-
     if debug:
         _setup_cli_debug()
 
+    if config.DISABLE_EVENTS:
+        return
+    ctx = click.get_current_context()
+    cmd = ctx.invoked_subcommand
+    publish_cmd_process = Process(target=_publish_cmd_as_analytics_event, args=(cmd,))
```

Comment:
do you think a Thread would be sufficient? i guess in that case we also need a way to pass a timeout to the underlying IO mechanism, which i guess is urllib underlying the requests library.

Reply:
Yep. I elected to use a Process here because there's no way to kill a thread externally. I considered passing a timeout down to the POST request, but thought that it would be cleaner to use a Process. Feel free to disagree!